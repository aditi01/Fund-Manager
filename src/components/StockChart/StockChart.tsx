import {
    Chart as ChartJS,
    CategoryScale,
    LinearScale,
    PointElement,
    LineElement,
    Title,
    Tooltip,
    Legend
  } from "chart.js";
  import { Line } from "react-chartjs-2";
  
  ChartJS.register(
    CategoryScale,
    LinearScale,
    PointElement,
    LineElement,
    Title,
    Tooltip,
    Legend
  );

interface Dataset {
  borderColor: string
  data: number[]
  fill: boolean
  label: string
}

interface StockData {
  labels: string[]
  datasets: Dataset[]
}

interface IProps {
  data: StockData
}

const StockChart = (props: IProps) => {
  const { data } = props

  const options = {
    responsive: true,
    scales: {
      x: {
        display: true,
        text: 'Date'
      },
      y: {
        title: {
          display: true,
          text: 'Price'
        }
      }
    },
  };

  return <Line data={data} options={options} />
};

export default StockChart;
