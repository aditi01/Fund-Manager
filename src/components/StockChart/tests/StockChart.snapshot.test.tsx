import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import StockChart from '../StockChart';

describe('StockChart', () => {
  const mockData = {
    labels: ['January', 'February', 'March', 'April', 'May', 'June', 'July'],
    datasets: [
      {
        label: 'Dataset 1',
        data: [65, 59, 80, 81, 56, 55, 40],
        fill: false,
        borderColor: 'rgba(75,192,192,1)',
      },
    ],
  };

  it('renders the chart with the given data and matches snapshot', () => {
    const { container } = render(<StockChart data={mockData} />);
    
    expect(container).toMatchSnapshot();
  });
});