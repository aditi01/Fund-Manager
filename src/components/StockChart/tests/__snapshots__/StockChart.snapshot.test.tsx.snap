// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockChart renders the chart with the given data and matches snapshot 1`] = `
<div>
  <canvas
    height="150"
    role="img"
    width="300"
  />
</div>
`;
